services/web/server/src/simcore_service_webserver/meta_iterations.py

```diff
@@ -0,0 +1,332 @@
+        cls, tag_name: str, *, return_none_if_fails: bool = False
+    ) -> Optional["ProjectIteration"]:
+        """Parses iteration info from tag name"""
+        try:
+            return cls.parse_obj(parse_iteration_tag_name(tag_name))
+        except ValidationError as err:
+            if return_none_if_fails:
+                log.debug("%s", f"{err=}")
+                return None
+            raise
+
+    def to_tag_name(self) -> str:
+        """Composes unique tag name for this iteration"""
+        return compose_iteration_tag_name(
+            repo_commit_id=self.repo_commit_id,
+            iter_index=self.iter_index,
+            total_count=self.total_count,
+            parameters_checksum=self.parameters_checksum,
+        )
+
+
+def compose_iteration_tag_name(
+    repo_commit_id: CommitID,
+    iter_index: int,
+    total_count: Union[int, str],
+    parameters_checksum: SHA1Str,
+) -> str:
+    """Composes unique tag name for iter_index-th iteration of repo_commit_id out of total_count"""
+    # NOTE: prefers to a json dump because it allows regex search?
+    # should be searchable e.g. get repo/*/*/*
+    # can do the same with json since it is a str!?
+    return (
+        f"iteration:{repo_commit_id}/{iter_index}/{total_count}/{parameters_checksum}"
+    )
+
+
+def parse_iteration_tag_name(name: str) -> Dict[str, Any]:
+    if m := re.match(
+        r"^iteration:(?P<repo_commit_id>\d+)/(?P<iter_index>\d+)/(?P<total_count>-*\d+)/(?P<parameters_checksum>.*)$",
+        name,
+    ):
+        return m.groupdict()
+    return {}
+
+
+# GET/CREATE iterations ------------------------------------------------------------
+
+
+async def get_or_create_runnable_projects(
+    request: web.Request,
+    project_uuid: ProjectID,
+) -> Tuple[List[ProjectID], List[CommitID]]:
+    """
+    Returns ids and refid of projects that can run
+    If project_uuid is a std-project, then it returns itself
+    If project_uuid is a meta-project, then it returns iterations
+    """
+
+    # FIXME: what happens if not
+
+    vc_repo = VersionControlForMetaModeling(request)
+    assert vc_repo.user_id  # nosec
+
+    # TODO:  handle UserUndefined and translate into web.HTTPForbidden
+    project: ProjectDict = await vc_repo.get_project(str(project_uuid))
+
+    project_nodes: Dict[NodeID, Node] = {
+        nid: Node.parse_obj(n) for nid, n in project["workbench"].items()
+    }
+
+    # init returns
+    runnable_project_vc_commits: List[CommitID] = []
+    runnable_project_ids: List[ProjectID] = [
+        project_uuid,
+    ]
+
+    # auto-commit
+    #   because it will run in parallel -> needs an independent working copy
+    repo_id = await vc_repo.get_repo_id(project_uuid)
+    if repo_id is None:
+        repo_id = await vc_repo.init_repo(project_uuid)
+
+    main_commit_id = await vc_repo.commit(
+        repo_id,
+        tag=f"auto:main/{project_uuid}",
+        message="auto-commit at get_or_create_runnable_projects",
```

Comment:
should you not put at least the time with HH:MM:SS in it? Would be easier to read through.